src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -175,7 +174,7 @@ DistributedTableSize(Oid relationId, char *sizeQuery)
 
 	ErrorIfNotSuitableToGetSize(relationId);
 
-	pgDistNode = heap_open(DistNodeRelationId(), AccessShareLock);
+	LockRelationOid(DistNodeRelationId(), AccessShareLock);
 
 	workerNodeList = ActivePrimaryNodeList();
```

Comment:
Good point, although I pushed it all the way down to WorkerNodeHash, which I think covers every entrypoint to the metadata cache.